tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11228,609 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i); 
+      } else if constexpr (std::is_same_v<T, bool>) {
+        // Compiler was very picky and wanted an explicit case for bools to
+        // avoid a warning about comparing bools with >.
+        if (vec1[i] != vec2[i]) {
+            vectorsMatch = false;
+            mismatchedIndexes.push_back(i); 
+        }
+      } else {
+        T diff = vec1[i] > vec2[i] ? vec1[i] - vec2[i] : vec2[i] - vec1[i];
+        if (diff > tolerance) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i);
+        }
+      }
+    }
+
+    // Print the mismatched indexes and their corresponding elements
+    if (!mismatchedIndexes.empty()) {
+        for (size_t index : mismatchedIndexes) {
+          std::wstringstream wss(L"");
+          wss << L"Mismatch at Index: " << index;
+          wss << L" Expected Value:" << vec1[index];
+          wss << L" Actual Value:" << vec2[index];
+          WEX::Logging::Log::Error(wss.str().c_str());
+        }
+    }
+
+    return vectorsMatch;
+}
+
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+T parseStringToNumber(std::wstring& str) {
+  if constexpr (std::is_same_v<T, uint16_t>) {
+    return static_cast<uint16_t>(std::stoul(str));
+  } else if constexpr (std::is_same_v<T, uint32_t>) {
+    return std::stoul(str);
+  } else if constexpr (std::is_same_v<T, uint64_t>) {
+    return std::stoull(str);
+  } else if constexpr (std::is_same_v<T, int16_t>) {
+    return static_cast<int16_t>(std::stol(str));
+  } else if constexpr (std::is_same_v<T, int32_t>) {
+    return std::stol(str);
+  } else if constexpr (std::is_same_v<T, int64_t>) {
+    return std::stoll(str);
+  } else if constexpr (std::is_same_v<T, float>) {
+    return std::stof(str);
+  } else if constexpr (std::is_same_v<T, double>) {
+    return std::stod(str);
+  } else if constexpr (std::is_same_v<T, bool>) {
+    bool boolVal = std::stod(str) > 0.0 ? true : false;
+    return boolVal ;
+  } else {
+    VERIFY_IS_TRUE(false, L"Unsupported type for parsing string to number.");
+  }
+}
+
+// Parse a vector of WEX::Common::Strings to a vector of numbers.
+// Expected string format is a comma separated list of numbers.
+// Strings with no commas could also be used for single numbers.
+// For example:
+// Three strings: ["1.0,2.0,3.0,4.0", "5.0,6.0,7.0,8.0", "9.0,10.0,11.0,12.0"]
+// would parse to 3 vectors of 4 elements each.
+// Or three strings ["1" , "2", "3"] will parse to 3 single elements vectors.
+// Vectors for a single element is a little weird, but it allows this helper
+// function to be very flexible.
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+std::vector<std::vector<T>> parseStringsToNumbers(const std::vector<WEX::Common::String>* input) {
+  std::vector<std::vector<T>> result = {};
+  if (input == nullptr || input->empty()) {
+      return result;
+  }
```

Comment:
Looks like your editor likes 4-space indents. The format checker is complaining about this too. Also no braces required 😄.
```suggestion
  if (input == nullptr || input->empty())
    return result;
```